unit_tests/MacroTranslTests.d

```diff
@@ -599,3 +599,41 @@ enum FOO = Bar.sizeof;
 D");
 
 }
+
+// Translate token concatenation operator.
+unittest
+{
+    assertTranslates(q"C
+#define FOO_CONCAT(prefix, name) prefix ## name
+C", q"D
+extern (C):
+
+extern (D) string FOO_CONCAT(T0, T1)(auto ref T0 prefix, auto ref T1 name)
+{
+    return prefix ~ name;
```

Comment:
It's possible to pass in integers to a macro like declared above. `~` will not work for integers.